pkg/messaging/topics.go

```diff
@@ -87,21 +85,55 @@ func ParseSubscribeTopic(topic string) (domainID string, chanID string, subtopic
 	return domainID, chanID, subtopic, nil
 }
 
-func ParseSubscribeSubtopic(subtopic string) (string, error) {
+func ParseSubscribeSubtopic(subtopic string) (parseSubTopic string, err error) {
 	if subtopic == "" {
 		return subtopic, nil
 	}
 
-	subtopic = strings.ReplaceAll(subtopic, "/", ".")
 	subtopic = strings.ReplaceAll(subtopic, "+", "*")
 	subtopic = strings.ReplaceAll(subtopic, "#", ">")
 
-	for _, ssnaChars := range subSubTopicNotAllowedChars {
-		if strings.Contains(subtopic, ssnaChars) {
-			return "", ErrMalformedSubtopic
+	subtopic, err = formatSubTopic(subtopic)
+	if err != nil {
+		return "", errors.Wrap(ErrMalformedSubtopic, err)
+	}
+
+	elems := strings.Split(subtopic, ".")
```

Comment:
Why don't we perform checks for invalidChars before the split? Also, consider using `strings.ContainsAny`.